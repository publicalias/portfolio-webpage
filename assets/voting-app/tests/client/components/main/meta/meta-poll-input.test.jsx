"use strict";

//local imports

const MetaPollInput = require("../../../../../scripts/client/components/main/meta/meta-poll-input");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { initTestEvent, initTestSnapshot, withDataList } = require("redux/tests/client-tests");
const { testMock } = require("redux/tests/meta-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(MetaPollInput);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(MetaPollInput));

//meta poll input

describe("MetaPollInput", () => {

  const testSnapshot = withDataList(initTestSnapshot(testShallow), [null, {
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    value: ""
  }]);

  const testChange = initTestEvent(testShallow, "change", {});

  const testSubmit = initTestEvent(testShallow, "click", {});

  it("should match snapshot", () => testSnapshot());

  it("should call handleChange on change", () => {

    const handleChange = jest.fn();

    return testChange(".qa-change-input", [null, { handleChange }], () => {
      testMock(handleChange, [{}]);
    });

  });

  it("should call handleSubmit on click", () => {

    const handleSubmit = jest.fn();

    return testSubmit(".qa-submit-input", [null, { handleSubmit }], () => {
      testMock(handleSubmit, [{}]);
    });

  });

});
