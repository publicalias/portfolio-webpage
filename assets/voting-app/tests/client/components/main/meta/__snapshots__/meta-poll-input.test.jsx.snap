// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaPollInput should match snapshot 1`] = `
<div
  className="c-meta-poll-options__input-box"
>
  <input
    className="c-meta-poll-options__input qa-change-input u-margin-right"
    maxLength="100"
    onChange={[MockFunction]}
    placeholder="New Option (Up to 20)"
    value=""
  />
  <button
    className="c-meta-poll-options__submit qa-submit-input"
    onClick={[MockFunction]}
  >
    Add
  </button>
</div>
`;
